dymos/run_problem.py

```diff
@@ -109,24 +112,33 @@ def run_problem(problem, refine=False, refine_iteration_limit=10, run_driver=Tru
 
         phases = find_phases(problem.model)
 
-        ref = PHAdaptive(phases)
+        ref = refinement_methods[refine_method](phases)
         with open(out_file, 'w+') as f:
 
             for i in range(refine_iteration_limit):
                 refine_results = check_error(phases)
 
-                ref.refine(refine_results)
-
-                for stream in f, sys.stdout:
-                    ref.write_iteration(stream, i, phases, refine_results)
-
                 refined_phases = [phase_path for phase_path in refine_results if
                                   phases[phase_path].refine_options['refine'] and
                                   np.any(refine_results[phase_path]['need_refinement'])]
 
                 if not refined_phases:
                     break
 
+                if refine_method == 'hp':
```

Comment:
The grid refinement loop should be independent of the refinement method used.  It looks like if we changed the `refine` method such that it took `refine_results` as well as the iteration as arguments, then everything done here could be pushed to the refine method of hp, while ph would just ignore the iteration number.